### PR TITLE
Use proc table for embedder APIs

### PR DESCRIPTION
**Pull request by bbrto21**
* Use the proc table like other platform implementations
* Use FlutterEngine as private memeber of FlutterTizenEngine
* Only FlutterTizenEngine is dependent on FlutterEngine

Signed-off-by: Boram Bae <boram21.bae@samsung.com>

### CODE REVIEW

**Comment by bbrto21** [comment]
I will make additional changes after #86 merged

**Comment by swift-kim** [comment]
As far as I know the only advantage of using proc table is improved testability. So there's no benifit for us because we don't have any tests. Still this change looks good.

**Comment by bbrto21** [comment]
Yes, you're right. I thought we would need it later but It is a simple change, so I did it right away 😄

**Comment by bbrto21** [comment]
@swift-kim Well... I have a question. it seems that our engine always has NDEBUG turned on both release and debug. 
It doesn't seem very important, but... is it the correct configuration?

**Comment by swift-kim** [comment]
@bbrto21

1. NDEBUG should be defined for release builds only:

   https://github.com/flutter/buildroot/blob/ef50dec1f85ee435e967505f460d2c0ce6fde58e/build/config/BUILD.gn#L84-L85

2. The `is_debug` flag depends on the `--unoptimized` build option, but not the `--runtime-mode` option - I don't know why:

   https://github.com/flutter-tizen/engine/blob/9f408ef240340dc9ce271a6cfe84d68e04cf977f/tools/gn#L114

   May be related to the change https://github.com/flutter/engine/pull/2669.

3. We don't set the option in the CI build:

   https://github.com/flutter-tizen/engine/blob/9f408ef240340dc9ce271a6cfe84d68e04cf977f/azure-pipelines.yml#L72-L82

So it's a bug. 😨 We need to fix our build config.

**Comment by bbrto21** [comment]
> So it's a bug.  We need to fix our build config.

@swift-kim @pwasowski2 
if it means that use ```--unoptimized``` for debug mode, I think that the symbol resolution will succeed

**Review by swift-kim — COMMENTED** [review]
Beautiful!

File: shell/platform/tizen/flutter_tizen_engine.h [hunk 1]
```diff
@@ -67,6 +68,8 @@ class FlutterTizenEngine : public TizenRenderer::Delegate {
 
   void InitializeRenderer();
 
+  void NotifyLowMemoryWarning();
```
[the remark this answers is not in the document]
Comment (bbrto21) [reply shell/platform/tizen/flutter_tizen_engine.h:71]: Should I arrange the functions in the order listed below?

GetPluginRegistrar
GetTextureRegistrar
InitializeRenderer
MarkExternalTextureFrameAvailable
OnOrientationChange
OnVsync
RegisterExternalTexture
RunEngine
SendPlatformMessage
SendPlatformMessageResponse
SendPointerEvent
SendWindowMetrics
SetPluginRegistrarDestructionCallback
SetWindowOrientation
StopEngine
UnregisterExternalTexture
UpdateLocales

File: shell/platform/tizen/flutter_tizen_engine.cc [hunk 2]
```diff
@@ -257,6 +265,52 @@ void FlutterTizenEngine::SetPluginRegistrarDestructionCallback(
...
+    const size_t message_size,
+    const FlutterDesktopBinaryReply reply,
+    void* user_data) {
+  FlutterPlatformMessageResponseHandle* response_handle = nullptr;
+  if (reply != nullptr && user_data != nullptr) {
+    FlutterEngineResult result =
+        embedder_api_.PlatformMessageCreateResponseHandle(
+            flutter_engine, reply, user_data, &response_handle);
+    if (result != kSuccess) {
+      std::cout << "Failed to create response handle\n";
```
[the remark this answers is not in the document]
Comment (bbrto21) [reply shell/platform/tizen/flutter_tizen_engine.cc]: Thanks!

File: shell/platform/tizen/flutter_tizen_engine.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (bbrto21) [reply shell/platform/tizen/flutter_tizen_engine.h:70]: Please review again

File: shell/platform/tizen/flutter_tizen_engine.h [hunk 3]
```diff
@@ -116,6 +134,12 @@ class FlutterTizenEngine : public TizenRenderer::Delegate {
   FlutterDesktopMessage ConvertToDesktopMessage(
       const FlutterPlatformMessage& engine_message);
   FlutterRendererConfig GetRendererConfig();
+  UniqueAotDataPtr LoadAotData(std::string aot_data_path);
+
+  FlutterEngineProcTable embedder_api_ = {};
+
+  // The Flutter engine instance.
+  FLUTTER_API_SYMBOL(FlutterEngine) flutter_engine;
```
[the remark this answers is not in the document]
Comment (bbrto21) [reply shell/platform/tizen/flutter_tizen_engine.h]: done!

File: shell/platform/tizen/flutter_tizen_engine.h [hunk 4]
```diff
@@ -116,6 +134,12 @@ class FlutterTizenEngine : public TizenRenderer::Delegate {
   FlutterDesktopMessage ConvertToDesktopMessage(
       const FlutterPlatformMessage& engine_message);
   FlutterRendererConfig GetRendererConfig();
+  UniqueAotDataPtr LoadAotData(std::string aot_data_path);
```
[the remark this answers is not in the document]
Comment (bbrto21) [reply shell/platform/tizen/flutter_tizen_engine.h]: Done!

**Review by swift-kim — APPROVED** [review]

File: shell/platform/tizen/flutter_tizen_engine.h [hunk 5]
```diff
@@ -66,25 +67,56 @@ class FlutterTizenEngine : public TizenRenderer::Delegate {
...
 
   void InitializeRenderer();
-
   bool RunEngine(const FlutterDesktopEngineProperties& engine_properties);
   bool StopEngine();
 
   // Returns the currently configured Plugin Registrar.
   FlutterDesktopPluginRegistrarRef GetPluginRegistrar();
-
   FlutterTizenTextureRegistrar* GetTextureRegistrar();
```
[the remark this answers is not in the document]
Comment (bbrto21) [reply shell/platform/tizen/flutter_tizen_engine.h]: Sorry! My mistake!

**Pull request merged**